tutorials/how_to/span_representations.md

```diff
@@ -0,0 +1,180 @@
+document lengths of upwards of 3000 tokens.
+
+In order to solve this problem, we need to be able to _prune_ spans as we go inside our
+model. There are several ways to do this:
+
+### Heuristically prune spans in your DatasetReader.
+
+We have added a utility method for enumerating all spans in a sentence, but excluding
+those which fulfil some condition based on the input text or any Spacy `Token` attribute.
+For instance, for co-reference, all spans which are mentions (spans which are 
+co-referent with _something_) never start or end with punctuation, because of the way
+the Onotonotes 5.0 dataset was created. This means that we can exclude any span which
+starts or ends with punctuation using a very simple python function:
+
+```
+from typing import List
+from allennlp.data.dataset_readers.dataset_utils import span_utils
+from allennlp.data.tokenizers.word_tokenizer import SpacyWordSplitter
+from allennlp.data.tokenizers.token import Token
+
+tokenizer = SpacyWordSplitter(pos_tags=True)
+sentence = tokenizer.split_words("This is a sentence.")
+
+def no_prefixed_punctuation(tokens: List[Token]) -> bool:
+    # Only include spans which don't start or end with punctuation.
+    return tokens[0].pos_ != "PUNCT" and tokens[-1].pos_ != "PUNCT"
+
+spans = span_utils.enumerate_spans(sentence,
+                                    max_span_width=3,
+                                    min_span_width=2,
+                                    filter_function=no_prefixed_punctuation)
+
+# 'spans' won't include (2, 4) or (3, 4) as these have
+# punctuation as their last element. Note that these spans
+# have inclusive start and end indices!
+assert spans == [(0, 1), (0, 2), (1, 2), (1, 3), (2, 3)]
+```
+
+There are other helpful functions in `allennlp.data.dataset_readers.dataset_utils.span_utils`,
+such as a function to convert between BIO labelings and span based representations.
+
+### Use a SpanPruner
+
+It's not always possible to prune spans before they enter your model. AllenNLP contains
+a `SpanPruner`, which allows you to prune spans based on a parameterised function which
+is trained end-to-end with the rest of your model.
+
+```
+import torch
+from torch.autograd import Variable
+from allennlp.modules import SpanPruner
+
+# Create a linear layer which will score our spans.
+linear_scorer = torch.nn.Linear(5, 1)
+pruner = SpanPruner(scorer=linear_scorer)
+
+# Here we'll create some spans from a random tensor of shape
+# (batch_size, num_spans, embedding_size). Typically this would
+# be the output of a SpanExtractor applied to some encoded representation
+# of a sentence, such as the output of an LSTM, or word embeddings.
+spans = Variable(torch.randn([3, 4, 5]))
+mask = Variable(torch.ones([3, 4]))
+
+# Ok, there's quite a bit to unpack here.
+# See below for a full explaination.
+pruned_embeddings, pruned_mask, pruned_indices, pruned_scores = pruner(spans, mask, num_spans_to_keep=3)
+```
+
+1. First, we've got our `pruned_embeddings`. 
```

Comment:
It's not immediately obvious that you're continuing here where the comment left off.  You might put a line like "A `SpanPruner` has four return values:" before this list.